Update branding to 3.0.1

dougbu opened the pull request:
- aspnet/AspNetCore-Internal#3153

dougbu committed:
Update branding to 3.0.1
- aspnet/AspNetCore-Internal#3153

dougbu committed:
Disable up-to-date check of package baselines

dougbu committed:
Correct and centralize a couple of conditions
- add `$(IgnorePackageBaselines)` property

dougbu committed:
Further centralize choice to opt out of servicing features for now
- no changes in known frameworks, targeting pack features, et cetera

dougbu commented:
```
F:\workspace\_work\1\s\src\Framework\ref\Microsoft.AspNetCore.App.Ref.csproj(177,5): error MSB3073: The command "tar -czf F:\workspace\_work\1\s\artifacts\installers\Release\aspnetcore-targeting-pack-3.0.1-ci.tar.gz ." exited with code 9009.
```
Either the targeting pack is not written anymore or its path changed. Both would be an unintended side effect of my changes.

BrennanConroy commented:
That's unrelated to this change. It's happening on a bunch of PRs. @JunTaoLuo and I think it might be an agent issue.

dougbu commented on eng/targets/ResolveReferences.targets:
```diff
@@ -36,7 +36,8 @@
     <UseLatestPackageReferences Condition=" '$(UseLatestPackageReferences)' == '' AND '$(IsImplementationProject)' != 'true' ">true</UseLatestPackageReferences>
-    <UseLatestPackageReferences Condition=" '$(UseLatestPackageReferences)' == '' AND '$(IsImplementationProject)' == 'true' AND ( '$(IsServicingBuild)' != 'true' OR '$(IsPackable)' == 'true' ) ">true</UseLatestPackageReferences>
+    <UseLatestPackageReferences
+        Condition=" '$(UseLatestPackageReferences)' == '' AND '$(IsImplementationProject)' == 'true' AND '$(IsPackable)' == 'true' ">true</UseLatestPackageReferences>
```
Remaining changes in this file just address nits. Removed of `'$(IsServicingBuild)' != 'true' OR` because that's already checked 2 lines above.

dougbu commented:
> @JunTaoLuo and I think it might be an agent issue.

Well, exit code 9009 means "file not found". If it's an agent issue, that would mean `tar` isn't found. @MattGal have the agents in NetCorePublic-Pool / BuildPool.Windows.10.Amd64.VS2019.Open moved to an **older** Windows build? (`tar` is now available by default but that's fairly recent.)

@BrennanConroy thanks, this makes more sense than my thought the targeting pack itself wasn't found. The command _creates_ the .tar.gz file from files in its current directory.

dougbu committed:
Disable building targeting pack
- `$(IsServicingBuild)` disabled at the moment

dougbu commented:
This is ready to go once `tar` is found on the agents again

dougbu committed:
Find or install Tar on CI
- work around dotnet/core-eng#7970
- install Git in repo if Tar can't be found in usual locations
- use found or installed Tar in Microsoft.AspNetCore.App.Ref project
  - copy into repo from wherever it's found
- add lots of `Write-Host` debugging

nit: clean up / comment on VS Code warnings about build.ps1

dougbu commented:
:up::date: to include a more complete dotnet/core-eng#7970 workaround
- may actually build this time 🤞

JunTaoLuo commented on Directory.Build.props:
```diff
@@ -86,7 +86,7 @@
 
     <!-- Used to only produce targeting pack installers/packages once per major.minor. -->
-    <IsTargetingPackBuilding Condition="'$(IsServicingBuild)' == 'true' AND '$(DotNetBuildFromSource)' != 'true'">false</IsTargetingPackBuilding>
+    <IsTargetingPackBuilding Condition="'$(AspNetCorePatchVersion)' == '0' AND '$(DotNetBuildFromSource)' != 'true'">false</IsTargetingPackBuilding>
```
Hmm well, we need to rebuild the targeting pack in 3.0.1 so this logic needs some finessing. We need to be able to control whether the targeting pack builds. We can use this as a default logic though.

JunTaoLuo commented on build.ps1:
```diff
@@ -328,10 +330,10 @@ if ($CI) {
-rm variable:global:_DotNetInstallDir -ea Ignore
-rm variable:global:_ToolsetBuildProj -ea Ignore
-rm variable:global:_MSBuildExe -ea Ignore
+Remove-Item variable:global:_BuildTool -ea Ignore
```
Why this change? Looks fine but I'm just curious.

dougbu replied:
Because aliases are bad in scripts and VS Code warns about them

JunTaoLuo reviewed (approved):
🥇 for coming up with a workaround to unblock us!

dougbu commented:
Success modulo a missing `New-Item -ItemType Directory .\.tools\ -ErrorAction SilentlyContinue` ❕ ❕

dougbu committed:
Create missing directory and simplify workaround slightly
